code/modules/xenoarchaeology/tools/xenoarchaeology_labeler.dm

```diff
@@ -0,0 +1,261 @@
+/*
+	Sticker for labeler, so we can label artifact's with their traits
+*/
+
+/obj/item/sticker/xenoartifact_label
+	icon = 'icons/obj/xenoarchaeology/xenoartifact_sticker.dmi'
+	icon_state = "sticker_star"
+	name = "artifact label"
+	desc = "An adhesive label, for artifacts."
+	do_outline = FALSE
+	roll_unusual = FALSE
+	///List of artifact traits we're labelling
+	var/list/traits = list()
+	///A special examine description built from the traits we have
+	var/examine_override = ""
+	///The original custom price of the item we're going to label
+	var/old_custom_price
+
+/obj/item/sticker/xenoartifact_label/old
+	name = "old artifact label"
+	color = "#bd812e"
+
+/obj/item/sticker/xenoartifact_label/old/build_stuck_appearance()
+	var/mutable_appearance/MA = mutable_appearance(sticker_icon || src.icon, sticker_icon_state || src.icon_state)
+	MA.color = color
+	return MA
+
+
+/obj/item/sticker/xenoartifact_label/Initialize(mapload, list/_traits)
+	ADD_TRAIT(src, TRAIT_ARTIFACT_IGNORE, GENERIC_ITEM_TRAIT)
+	//Setup traits & examine desc
+	traits = _traits.Copy()
+	if(length(traits))
+		examine_override = "Traits:"
+		for(var/datum/xenoartifact_trait/T as() in traits)
+			examine_override = "[examine_override]\n	- [initial(T.label_name)]"
+	//Setup a random appearance
+	icon_state = "sticker_[pick(list("star", "box", "tri", "round"))]"
+	sticker_icon_state = "[icon_state]_small"
+	return ..()
+
+/obj/item/sticker/xenoartifact_label/examine(mob/user)
+	. = ..()
+	. += examine_override
+
+/obj/item/sticker/xenoartifact_label/afterattack(atom/movable/target, mob/user, proximity_flag, click_parameters)
+	//If you somehow make traits start working with mobs, remove this isliving() check
+	if(!isliving(target) && (locate(/obj/item/sticker/xenoartifact_label) in target.contents))
+		to_chat(user, "<span class='watning'>[target] already has a label!</span>")
+		return
+	. = ..()
+	if(!can_stick(target) || !proximity_flag)
+		return
+	//Set custom price with the artifact component
+	var/datum/component/xenoartifact/artifact = target.GetComponent(/datum/component/xenoartifact)
+	if(artifact)
+		old_custom_price = target.custom_price
+		//Build list of artifact's traits
+		var/list/artifact_traits = list()
+		for(var/i in artifact.artifact_traits)
+			for(var/datum/xenoartifact_trait/T as() in artifact.artifact_traits[i])
+				artifact_traits += T
+		//Compare them to ours
+		for(var/datum/xenoartifact_trait/T as() in traits)
+			if(locate(T) in artifact_traits)
+				target.custom_price *= XENOA_LABEL_REWARD
+			else
+				target.custom_price *= XENOA_LABEL_PUNISHMENT
+	RegisterSignal(target, COMSIG_PARENT_EXAMINE, PROC_REF(parent_examine))
+
+/obj/item/sticker/xenoartifact_label/attack_hand(mob/user)
+	if(sticker_state == STICKER_STATE_STUCK)
+		UnregisterSignal(loc, COMSIG_PARENT_EXAMINE)
+	//Set custom price back
+	var/datum/component/xenoartifact/artifact = loc.GetComponent(/datum/component/xenoartifact)
+	if(artifact)
+		loc.custom_price = old_custom_price
```

Comment:
loc.custom_price??? I know where this came from, but this is not a good standard. get the variable `custom_price` from a typeholder variable correctly.

Reply:
![image](https://github.com/user-attachments/assets/65551991-f73e-46e8-a47a-1f0a5d5de903)
![image](https://github.com/user-attachments/assets/7a6dc7a8-bd8a-47fe-9af8-c91394d65dbf)

Reply:
Note that this is used by vending machines and not supply

Reply:
> Note that this is used by vending machines and not supply

Yeah, I know. Some of the other Xenoarch code borrows it though, to get pricing for cargo stuff, just so it's all unified.